File: 1497. Check If Array Pairs Are Divisible by k.java
class Solution {
    public boolean canArrange(int[] arr, int k) {
        Map<Integer, Integer> map = new HashMap<>();
        int current_pairs = 0;
        for(int val : arr){
            int remain = val % k;
            if (remain < 0){
                remain += k;
            }
            int count = map.getOrDefault(k - remain, 0);
            if ( count > 0){
                map.put(k - remain, --count);
                current_pairs++;
            }else{
                int key_count = map.getOrDefault(remain, 0);
                map.put(remain, key_count + 1);
            }
        }
        if(map.getOrDefault(0, 0) / 2 + current_pairs == arr.length / 2){
            return true;
        }else{
            return false;
        }
        
    }
}

File: 1497. Check If Array Pairs Are Divisible by k.py
class Solution:
    def canArrange(self, arr: List[int], k: int) -> bool:
        count = collections.defaultdict(int)
        for val in arr:
            remain = val % k
            if remain < 0:
                remain += k
            if count[k - remain] > 0:
                count[k - remain] -= 1
            else:
                count[remain] += 1
        if sum(count.values()) == count[0] and count[0] % 2 == 0:
            return True
        else:
            return False